### PR TITLE
Allow match field in enrich fields

### PR DESCRIPTION
**Pull request by dnhatn**
It's perfectly fine for the match field of an enrich policy to be included in the enrich fields. However ESQL enrich consistently fails on such an enrich policy because it mistakenly excludes the match field from the enrich mapping during resolution.

### CODE REVIEW

**Review by luigidellaquila — APPROVED** [review]
LGTM, thanks!

File: x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/esql/62_extra_enrich.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  - match: { columns.1.type: "integer" }
+
+  - length: { values: 1 }
+  - match: { values.0.0: "engineering" }
+  - match: { values.0.1: 1024 }
+
+  - do:
+      esql.query:
+        body:
+          query: 'ROW name="sales" | ENRICH departments-policy ON name WITH department=name | WHERE name=department | KEEP name, department | LIMIT 10'
```
Comment (luigidellaquila) [inline x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/esql/62_extra_enrich.yml:60]: `WHERE name=department` should be `WHERE name==department`

**Review by astefan — COMMENTED** [review]
I worked on the IndexResolution and field_caps field names to be used and I would like to understand more about the problem this PR is fixing, since that code is making use of enrich policies field names.
There is no bug link mentioned in the PR description, nor there is a failing query or exception. I am seeing the queries in the yml file and I tested those, but I don't truly understand the issue this PR is fixing.

**Comment by dnhatn** [comment]
@astefan Sure, if you run the new YAML test without the fix in the Analyzer, you would hit the error `{"type":"verification_exception","reason":"Found 1 problem\nline 1:26: Enrich field [name] not found in enrich policy [departments-policy]`. I found this issue and worked on this fix. No need for an issue for such a case. The PR description explains that ESQL enrich doesn't work with policies whose match field is included in their enrich fields. Please let me know if you need anything else to understand the fix.

**Comment by astefan** [comment]
Ok, now it makes sense.
The specific case this is covering is
```
                "match_field": "geo.dest",
                "enrich_fields": ["geo.dest", "population"]
```
where both `match_field` and `enrich_fields` have `geo.dest`.

**Review by astefan — APPROVED** [review]
LGTM

**Comment by brianseeders** [comment]
@dnhatn looks like this didn't get backported and won't be released tomorrow. Is that an issue?

**Comment by dnhatn** [comment]
@brianseeders Thanks for ping. I've opened the backport PR; however, it's fine if it doesn't make the release train tomorrow.

**Pull request merged**